### PR TITLE
Python 3 compatability

### PR DESCRIPTION
**Pull request by aholmberg**
Added some unit tests for regression coverage, updated to work with Python 3.
(ping #10)

### CODE REVIEW

**Comment by aholmberg** [comment]
Unit tests run in both 2 & 3
```
Name                  Stmts   Miss  Cover
-----------------------------------------
puresasl                 17      0   100%
puresasl.client          52      0   100%
puresasl.mechanisms     188     11    94%
-----------------------------------------
TOTAL                   257     11    96%
```

Also manually tested GSSAPI mechanism with DataStax Enterprise

File: test-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+nose
+mock<=1.0.1
```
Comment (thobbs) [inline test-requirements.txt]: What's the reason for the version restriction?  (I seem to recall an incompatibility with ``six`` in some versions, or something like that)

Comment (aholmberg) [inline test-requirements.txt]: They dropped Python 2.6 compatibility.
https://github.com/testing-cabal/mock/commit/8a2e8709502c30e2dd0838125bc5c34fec35f451
However, I see that they later returned it, so this is probably not necessary anymore. We would have to somehow exclude 1.1.0..1.1.4.
https://github.com/testing-cabal/mock/commit/a6367a9a2b6166d7d032ec91288294ec47177649

Comment (thobbs) [inline test-requirements.txt]: Thanks.  I'm fine with the current restriction for now.

**Comment by thobbs** [comment]
Merged.  Thanks for the nice work, as usual!

**Comment by aholmberg** [comment]
Thanks! Are you planning a release any time soon, or do we need to have py3 users install from here? No big deal either way -- just looking to set expectations.

**Pull request merged**